packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -837,6 +985,44 @@ function rerenderReducer<S, I, A>(
   return [newState, dispatch];
 }
 
+// @TODO this is a completely broken bailout implementation. figure out how
+// dispatchAction and updateReducer really work and implement something proper
+function bailoutReducer(hook, renderExpirationTime): boolean {
```

Comment:
going to take a pass as actually supporting proper expiration time management next here and elsewhere. Would love some advice on writing tests in react to fully exercise various concurrent edge cases so I can verify correctness

Reply:
Just to set expectations, @acdlite is planning a pretty invasive set of changes to expiration times

Reply:
Thanks @gaearon, I talked with Sebastian on Monday and he let me know the same and that this PR won't get a lot of attention for the near future given your more immediate plans. That's all good. @acdlite if you are willing to catch me up in the future when you know what it might look like I'd love to hear about it